pyarr/models/common.py

```diff
@@ -1,25 +1,23 @@
+from dataclasses import dataclass
 from enum import Enum
 
-import enum_tools.documentation
 
-enum_tools.documentation.INTERACTIVE = True
-
-
-@enum_tools.documentation.document_enum
-class PyarrSortDirection(str, Enum):
+@dataclass(order=True)
+class PyarrSortDirection(Enum):
```

Comment:
Do these still need to be `Enum`?

Reply:
Yes I tried it without but then get the following type error:

```
pyarr/readarr.py:90: error: Incompatible default for argument "author_monitor" (default has type "str", argument has type "ReadarrAuthorMonitor")  [assignment]
```

Reply:
After some testing, it seems that this can be done two ways: 

```
@dataclass(order=True)
class LidarrArtistMonitor(str):
    """Lidarr Monitor types for an artist music"""

    ALL_ALBUMS = "all"
```


```
@dataclass(order=True)
class LidarrArtistMonitor:
    """Lidarr Monitor types for an artist music"""

    @property
    def ALL_ALBUMS():
        return "all"
```

Reply:
Well the issue is the type hints here. `author_monitor: ReadarrAuthorMonitor` means that auth_monitor is expecting a class value, but the attribute `ReadarrAuthorMonitor.ALL` is a string.

I think suggesting to use strictly dataclasses for this on Python <3.11 may have been a mistake on my part. Look at option 3 for the best solution for this situation.

It looks like the options either are:

1) Set the attribute type to string...
   ```python
    def _book_json(
        ... ,
        author_monitor: str = ReadarrAuthorMonitor.ALL,
    ) -> JsonObject:
   ```
   - Which would allow for users to hard-code auth_monitor values rather than using our type objects, if needed.
   - The down side is this will not type hint to make sure the `auth_monitor` value is actually contained within `ReadarrAuthorMonitor`

2) Create `LidarrArtistMonitorType` and `LidarrArtistMonitor` separately
   ```python
   LidarrArtistMonitorType = typing.Literal["foo", "bar"]

   @dataclass
   class ReadarrAuthorMonitor:
       FOO = "foo"
       BAR = "bar"
   ```
   - `LidarrArtistMonitorType` would be the type hint, and `LidarrArtistMonitor` would be the dataclass used by users to access values.
   - But unfortunately, there is no way of doing this that doesn't require duplicating all the string options into both `LidarrArtistMonitorType` and `LidarrArtistMonitor`. This is something that will be fixed in Python 3.11, which isn't widespread yet.

3) Declare `LidarrArtistMonitorType` as a `typing.Literal` and provide no dataclass/Enum method of fetching the values.
   ```python
   LidarrArtistMonitorType = typing.Literal['foo', 'bar']
   ```
   - In my opinion, this is the simplest and best solution. 
   - This solution can be expanded to encompass option 2 whenever Python 3.11 becomes out minimum version. That is, if we ultimately want the pyarr to be more object-oriented.

4) Using an `Enum`
   ```python
   class ReadarrAuthorMonitor(Enum):
       ALL = "all"
       FUTURE = "future"
       MISSING = "missing"
       EXISTING = "existing"
       FIRST_BOOK = "first"
       LATEST_BOOK = "latest"
       NONE = "none"
   ```
   - Your old implementation inherits from both `str` and `Enum`, however, inheriting from `str` is definitely improper.
   - This however, requires users to use `ReadarrAuthorMonitor.ALL.value` which is kind of ugly.
   - And it doesn't properly type check things from my testing, so not worth using.

Reply:
Yes option 3 may work, Hopefully I can get sphinx to document the literal types options then at least people can lookup the available options as they wont be listed without the enum/dataclass.